ddsrouter_test/compose/scripts/execute_and_validate_subscriber.py

```diff
@@ -124,53 +136,77 @@ def _subscriber_parse_output(stdout, stderr):
     :param data: Process stdout
     :return: List of received messages
     """
-    regex = re.compile(r'^Message\sHelloWorld\s+\d+\sRECEIVED$')
-    lines = stdout.splitlines()
-    filtered_data = [line for line in lines if regex.match(line)]
+
+    msgs_regex = re.compile(r'^Message\sHelloWorld\s+\d+\sRECEIVED$')
+    match_regex = re.compile(r'^Subscriber matched \[.+\].$')
+    unmatch_regex = re.compile(r'^Subscriber unmatched \[.+\].$')
+
+    filtered_data = {'messages': [], 'matches': 0, 'unmatches': 0}
+
+    for line in stdout.splitlines():
+        if msgs_regex.match(line):
+            filtered_data['messages'].append(line)
+
+        elif match_regex.match(line):
+            filtered_data['matches'] += 1
+
+        elif unmatch_regex.match(line):
+            filtered_data['unmatches'] += 1
 
     return filtered_data, stderr
 
 
 def _subscriber_get_retcode_validate(
         samples):
     if samples == 0:
-        def accept_timeout(retcode):
-            return (
-                retcode == validation.ReturnCode.SUCCESS
-                or retcode == validation.ReturnCode.TIMEOUT)
-        return accept_timeout
-    else:
-        return validation.validate_retcode_default
+        return lambda retcode: retcode == validation.ReturnCode.SUCCESS or \
+                               retcode == validation.ReturnCode.TIMEOUT
+
+    return validation.validate_retcode_default
 
 
 def _subscriber_validate(
         stdout_parsed,
         stderr_parsed,
         samples,
         duplicates_allow,
-        transient):
+        transient,
+        n_matches,
+        n_unmatches):
 
     # Check default validator
-    ret_code = validation.validate_default(stdout_parsed, stderr_parsed)
+    ret_code = validation.validate_default(stdout_parsed['messages'], stderr_parsed)
 
     if duplicates_allow != -1:
-        duplicated_n = len(find_duplicates(stdout_parsed))
+        duplicated_n = len(find_duplicates(stdout_parsed['messages']))
         if duplicated_n > duplicates_allow:
             log.logger.error(
                 f'{duplicated_n} duplicated messages found. '
                 f'Maximum allowed {duplicates_allow}.')
             return validation.ReturnCode.NOT_VALID_MESSAGES
 
     if transient:
-        if not check_transient(stdout_parsed):
+        if not check_transient(stdout_parsed['messages']):
             log.logger.error('Transient messages incorrect reception.')
             return validation.ReturnCode.NOT_VALID_MESSAGES
 
-    if len(stdout_parsed) != samples:
-        log.logger.error(f'Number of messages received: {len(stdout_parsed)}. '
+    if len(stdout_parsed['messages']) != samples:
+        log.logger.error(f'Number of messages received: {len(stdout_parsed["messages"])}. '
                          f'Expected {samples}')
         return validation.ReturnCode.NOT_VALID_MESSAGES
 
+    if n_matches is not None and stdout_parsed['matches'] != n_matches:
```

Comment:
I would apply this same logic to `samples`. If not specified, it should not be considered. The executable shouldn't get samples argument.
In that case, timeout should be accepted if `samples` isn't specified or if it is equal to zero.

Reply:
Check `samples is not None` in line 192.